### DISCUSSION TITLE
Understanding charge current curve

### DISCUSSION THREAD

**Discussion by SergioRius** [Louisvdw/dbus-serialbattery#944] [Q&A]
I have two banks of 280Ah in 18s. I first configured the driver for the acceptable/safe max charging current for the total amount of cells. That resulted in an excess of CCL and an nonexistent taper.

Then I applied the following configuration:
```
[DEFAULT]

; If you want to add custom values/settings, then check the values/settings you want to change in "config.default.ini"
; and insert them below to persist future driver updates.

MAX_BATTERY_CHARGE_CURRENT    = 60.0
MAX_BATTERY_DISCHARGE_CURRENT = 140.0

MIN_CELL_VOLTAGE   = 2.930
MAX_CELL_VOLTAGE   = 3.450
FLOAT_CELL_VOLTAGE = 3.400

SOC_RESET_VOLTAGE  = 3.570

; No se aplica en modo Linear
; MAX_VOLTAGE_TIME_SEC = 3600
; SOC_LEVEL_TO_RESET_VOLTAGE_LIMIT = 90

SOC_CALCULATION = True
; SOC_RESET_CURRENT = 7
SOC_RESET_TIME = 180


; --------- Cell Voltage Current limitation (affecting CCL/DCL) ---------

; Charge current control management referring to cell-voltage enable (True/False).
CCCM_CV_ENABLE = True
; Discharge current control management referring to cell-voltage enable (True/False).
; !!! Dejar Activo solo porque la reducción está por debajo del voltaje admisible
DCCM_CV_ENABLE = True

; Set steps to reduce battery current
; The current will be changed linear between those steps if LINEAR_LIMITATION_ENABLE is set to True
CELL_VOLTAGES_WHILE_CHARGING   = 3.55, 3.50, 3.45, 3.40, 3.39, 3.35
MAX_CHARGE_CURRENT_CV_FRACTION =    0, 0.01,  0.3,  0.5,    1,    1

CELL_VOLTAGES_WHILE_DISCHARGING   = 2.70, 2.80, 2.90, 3.00, 3.10
MAX_DISCHARGE_CURRENT_CV_FRACTION =    0,  0.1,  0.3,  0.5,    1


; --------- Cell Voltage limitation (affecting CVL) ---------

CVL_ICONTROLLER_MODE = True



; --------- Temperature limitation (affecting CCL/DCL) ---------

TEMPERATURE_LIMITS_WHILE_CHARGING = 0,   2,   5,  10, 20, 35,  40, 55
MAX_CHARGE_CURRENT_T_FRACTION     = 0, 0.1, 0.2,   1,  1,  1, 0.4,  0

TEMPERATURE_LIMITS_WHILE_DISCHARGING = -20,   0,   5,  10, 45, 55
MAX_DISCHARGE_CURRENT_T_FRACTION     =   0, 0.2, 0.3,   1,  1,  0


; --------- SOC limitation (affecting CCL/DCL) ---------

CCCM_SOC_ENABLE = False
DCCM_SOC_ENABLE = False

BMS_TYPE = Jkbms
```

The objective was to somewhat approach to the Victrons charge profile and allow for a tapered end of charge. But the results are not understandable to me. This is screenshot of last nights scheduled charge.
![imagen](https://github.com/Louisvdw/dbus-serialbattery/assets/1693727/ab7f1f5a-9b96-4f83-aac6-04ca96025927)
Until 01:45 the CCL (in green) is flat at 79A. Then with only 60.65v it reduces to 57A. This is the part that I don't understand.
Then at 61.24 it starts to taper to 18.3A. That is a high end but correct to me.
But then, 10min later it starts increasing current while at max voltage level and at the end of the charge, instead of reducing, it escalates again to 80A.

I can't understand this behavior, while the charge curve of the Victron system (in yellow) seems more logical to me.
There's no float in this CCL and then absorption is inexistent.

Can someone tell me if this is expected or what I'm doing wrong in my configuration?
What should be the recommended settings for a 560Ah system?

PS: Absorption is 62.1v, float is 61.2v

**Comment by snow78** [comment]
are you feeding in excessive DC power to grid? if so, this disables basically CCL. DC Feed in basically tells MPPT to produce at max. and Victron gear will feed in..... but Battery takes as much as it can before. you need to use CVL
